### PR TITLE
package-manager: always set Unhealthy() message

### PR DESCRIPTION
**Pull request by sttts**
### Description of your changes

Add condition messages to the package manager resources.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Added or updated unit **and** E2E tests for my change.
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~~Added `backport release-x.y` labels to auto-backport this PR, if necessary.~~
- [ ] ~~Opened a PR updating the [docs], if necessary.~~

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

### CODE REVIEW

File: test/e2e/funcs/feature.go [hunk 1]
```diff
@@ -219,7 +219,11 @@ func ResourcesHaveConditionWithin(d time.Duration, dir, pattern string, cds ...x
 			_ = fieldpath.Pave(u.Object).GetValueInto("status", &s)
 
 			for _, want := range cds {
-				if !s.GetCondition(want.Type).Equal(want) {
+				got := s.GetCondition(want.Type)
```
Comment (turkenh) [inline test/e2e/funcs/feature.go:222]: Same as https://github.com/crossplane/crossplane/pull/4584#discussion_r1325368954

File: internal/controller/pkg/revision/reconciler.go [hunk 2]
```diff
@@ -549,40 +556,45 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
...
 	// Lint package using package-specific linter.
 	if err := r.linter.Lint(pkg); err != nil {
-		pr.SetConditions(v1.Unhealthy())
+		log.Debug(errLintPackage, "error", err)
+
+		err = errors.Wrap(err, errLintPackage)
+		pr.SetConditions(v1.Unhealthy().WithMessage(err.Error()))
 		_ = r.client.Status().Update(ctx, pr)
 
+		r.record.Event(pr, event.Warning(reasonLint, err))
```
Comment (turkenh) [inline internal/controller/pkg/revision/reconciler.go:577]: Do you think if we have too much of similar lines that could be refactored into a dedicated function?

**Review by turkenh — APPROVED** [review]
Thanks @sttts 🙌

**Comment by sttts** [comment]
> I would suggest dropping the change https://github.com/crossplane/crossplane/pull/4610#discussion_r1325377039 before merging this PR, though.

Don't think it makes sense to compare messages in e2e. They might differ by some details, and adding a predicate to ignore the message is overkill. Am happy to make that clearer in the godocs.

**Comment by sttts** [comment]
Actually, I suspect some message comparison somewhere in the code as this PR consistently fails e2e.

File: test/e2e/funcs/feature.go (hunk 1 above)
Comment (sttts) [reply test/e2e/funcs/feature.go:222]: Updated. Ptal. Now it's unconditional, documented and the dev gets an error if a message is set in the comparison conditions.

File: internal/controller/pkg/revision/reconciler.go (hunk 2 above)
Comment (sttts) [reply internal/controller/pkg/revision/reconciler.go:577]: They are not really equal but differ in details. Also this is preexisting in the code. I have only moved a few lines consistently to make sense.

File: test/e2e/funcs/feature.go (hunk 1 above)
Comment (turkenh) [reply test/e2e/funcs/feature.go:226]: This might break some existing e2e tests, did you check that?

Comment (sttts) [reply test/e2e/funcs/feature.go:226]: Actually, you are right. It could break something :) Checking.

Comment (sttts) [reply test/e2e/funcs/feature.go:226]: Double checked. We should be fine.

**Comment by lsviben** [comment]
Is the last [commit](https://github.com/crossplane/crossplane/pull/4610/commits/ce2c28d1ebb9beaded1a9cb80569f7f74deb6a75) with making the ownership graph in the e2e tests supposed to be here? Not sure if on purpose or a testing leftover, feels out of place, although it seems useful

**Comment by sttts** [comment]
It's kind of intentional, see https://github.com/crossplane/crossplane/pull/4594. I needed it to understand flakes here.

**Comment by lsviben** [comment]
> It's kind of intentional, see #4594. I needed it to understand flakes here.

I see, looks super useful. Ill review that one after this to push it along

**Review by lsviben — APPROVED** [review]
LGTM, thanks @sttts !

**Pull request merged**